scripts/build-probe-binaries

```diff
@@ -53,26 +53,42 @@ else
 	PROBE_REPO_NAME=$(echo $PROBE_NAME | cut -f1 -d-)
 fi
 
-function update_code_for {
-	repo=$1
-	if [ ! -d $repo ]; then
-		git clone git@github.com:draios/$repo.git
+function checkout_code_for {
+    	repo=$1
+
+	IFS='/' read -ra parts <<< $repo
+
+	local_dir=${parts[1]}
+
+	if [ -d $local_dir ]; then
```

Comment:
```
git clean -d -f -x
git reset --hard
```
These commands below are used to set a repo back to exactly what's on the remote. Let's move them into a function and re-use it here so we avoid doing a full clone each time the builder runs.

Reply:
The motivation for this PR was actually to do a full clone every time the builder runs. It came up because we wanted to change the repo for the "falco" directory to be draios/oss-falco instead of draios/falco (which actually ends up being falcosecurity/falco, and won't have our tags on it), and I didn't want to ask ops to move the artifact out of the way.